internal/commands/push/command.go

```diff
@@ -115,7 +129,7 @@ func (cmd *Command) Handler(profile *cli.Profile, ui terminal.UI, clients cli.Cl
 		}
 	}
 
-	if len(appDiffs) == 0 && !cmd.inputs.IncludeDependencies && hostingDiffs.Size() == 0 {
+	if len(appDiffs) == 0 && (!cmd.inputs.IncludeDependencies || !dependenciesDiff.HasChanges()) && hostingDiffs.Size() == 0 {
```

Comment:
couldn't something like this suffice as a check?  (the usage of `IncludeDependencies` here was a stopgap b/c we had otherwise no data relevant to how the dependences had changed)
```suggestion
	if len(appDiffs) == 0 && dependenciesDiffs.Size() == 0 && hostingDiffs.Size() == 0 {
```

[naming nit] would call it `dependenciesDiffs`, or switch the other two to be singule with `*Diff`...